spec/loaders/abstract-loader-shared-behavior.coffee

```diff
@@ -303,6 +303,29 @@ registerSharedBehavior "AbstractLoaderSharedBehavior", (sharedContext) ->
           loader.setup(opts)
           notFound(loader, opts)
 
+      context 'when optional fields have been requested but the fields arent on all the tasks', ->
+        beforeEach ->
+          opts.optionalFields = ['test_field']
+          taskOne.set('test_field', 'fake value')
+          loader.storageManager.storage('tasks').add([taskOne, taskTwo])
+
+        it 'returns false and does not call #_onLoadSuccess', ->
+          loader.setup(opts)
+          loader._checkCacheForData()
+          expect(loader._onLoadSuccess).not.toHaveBeenCalled()
+
+      context 'when optional fields have been requested and the fields are already on the tasks', ->
+        beforeEach ->
+          opts.optionalFields = ['test_field']
+          taskOne.set('test_field', 'fake value for one')
+          taskTwo.set('test_field', 'fake value for two')
+          loader.storageManager.storage('tasks').add([taskOne, taskTwo])
+
+        it 'returns true and does not calls #_onLoadSuccess', ->
```

Comment:
Totally didn't read this description ಠ_ಠ. Same as below for testing the return value.